### PR TITLE
Feat/deprecate vault interfaces

### PR DESCRIPTION
**Pull request by alcueca**
Deprecate the vault-interfaces repo and bring the interfaces into vault-v2

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,21 +1,15 @@
...
 	url = https://github.com/makerdao/dss-interfaces
-[submodule "packages/foundry/lib/yieldspace-tv"]
-	path = packages/foundry/lib/yieldspace-tv
-	url = https://github.com/yieldprotocol/yieldspace-tv
 [submodule "packages/foundry/lib/forge-std"]
 	path = packages/foundry/lib/forge-std
 	url = https://github.com/foundry-rs/forge-std
+[submodule "packages/foundry/lib/yieldspace-tv"]
+	path = packages/foundry/lib/yieldspace-tv
+	url = https://github.com/yieldprotocol/yieldspace-tv
```
Comment (bbonanno) [inline .gitmodules:15]: So git resolves the circular dependency?

**Comment by bbonanno** [comment]
this also adds a lot of Uniswap contracts, ignoring scope of the PR, wouldn't it be better to use a foundry dependency when possible?

**Review by devtooligan — APPROVED** [review]
lgtm after addressing @bbonanno's questions

File: .gitmodules (hunk 1 above)
Comment (alcueca) [reply .gitmodules:15]: The circular dependencies are resolved with submodules, npm and remappings. I'm not very clear on how I managed to do it now.

Comment (alcueca) [reply .gitmodules:15]: Ok, so looking back into it, for foundry to work I do forge install and a remapping.

For hardhat, I do npm install.

**Pull request merged**